### PR TITLE
Resolves 10. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database so the app can learn how often I buy different items.

### LINKED ISSUE #10: 10. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database so the app can learn how often I buy different items.
AC:

* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

The script at `/src/lib/estimates.js` should be used to calculate the next estimated purchase interval.

### PR DESCRIPTION
**Pull request by wlcreate**
## Description

This code mainly deals with how we are writing to the Firebase database. When a user adds an item to their shopping list we are also recording the ```numberOfPurchases``` to keep track of how many times a user has purchased the item. Now when a user checks an item off their list these three things are updated:
*  ```daysToPurchase```, which based on ```estimates.js``` is the ```lastEstimate``` and is the return of ```calculateEstimate```
* ```lastPurchasedDate```, which updates to the time the user checks the item off
* ```numberOfPurchases``` , which increases by 1 each time the user checks the item off

## Related Issue

Closes #10 

## Acceptance Criteria

* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

Screen Shot 2021-02-04 at 11 08 00 AM

## Testing Steps / QA Criteria
* switch to our branch ```WL-JS-estimate-next-purchase-date```
* add item to list
* go to Firebase and change the ```lastPurchasedDate``` of the item to before the date you are testing
* check off the same item
* in Firebase you should see the item's ```daysToPurchase```, ```lastPurchasedDate```, and ```numberOfPurchases``` updated

### CODE REVIEW

File: src/components/AddItem.js [hunk 1]
```diff
@@ -24,25 +24,26 @@ const AddItem = () => {
...
-      alert('Item already on shoppping list');
+      alert("Item already on shoppping list");
       return;
     }
-    db.collection(localStorage.getItem('token')).add({
+    db.collection(localStorage.getItem("token")).add({
       itemName: groceryItem,
       daysToPurchase: daysToPurchase,
       lastPurchasedDate: null,
+      numberOfPurchases: 0,
```
Comment (jessicasalbert) [inline src/components/AddItem.js:34]: It looks like we are still having issues with competing linters for the quotes thing. This is the only line that changed in this file - adding a 'numberOfPurchases' key to the db for each item.

**Review by Maeesha-Rahman — APPROVED** [review]
Awesome work! Love the specific variable names. It seems to be working as expected based on the Testing Steps / QA Criteria. Approved, great work!! :smiley::star2::star2:

File: src/components/AddItem.js (hunk 1 above)
Comment (thekidnamedkd) [reply src/components/AddItem.js:34]: Maybe we need to manually change all our .prettierrc files on our next dev branches this coming week before we push up?

**Review by thekidnamedkd — APPROVED** [review]
This looks great. I think y'all hit the AC. I'm thinking per some Slack conversations, we should maybe start to part this out into some kind of Item logic separate from any list rendering concerns. Let's discuss! 🍿

**Review by MelinaMejia95 — APPROVED** [review]

**Pull request merged**